packages/flutter/lib/src/widgets/layout_builder.dart

```diff
@@ -11,56 +11,33 @@ import 'framework.dart';
 /// The signature of the [LayoutBuilder] builder function.
 typedef LayoutWidgetBuilder = Widget Function(BuildContext context, BoxConstraints constraints);
 
-/// Builds a widget tree that can depend on the parent widget's size.
-///
-/// Similar to the [Builder] widget except that the framework calls the [builder]
-/// function at layout time and provides the parent widget's constraints. This
-/// is useful when the parent constrains the child's size and doesn't depend on
-/// the child's intrinsic size. The [LayoutBuilder]'s final size will match its
-/// child's size.
-///
-/// {@youtube 560 315 https://www.youtube.com/watch?v=IYDVcriKjsw}
-///
-/// If the child should be smaller than the parent, consider wrapping the child
-/// in an [Align] widget. If the child might want to be bigger, consider
-/// wrapping it in a [SingleChildScrollView].
-///
-/// See also:
-///
-///  * [Builder], which calls a `builder` function at build time.
-///  * [StatefulBuilder], which passes its `builder` function a `setState` callback.
-///  * [CustomSingleChildLayout], which positions its child during layout.
-class LayoutBuilder extends RenderObjectWidget {
-  /// Creates a widget that defers its building until layout.
-  ///
-  /// The [builder] argument must not be null.
-  const LayoutBuilder({
+/// The signature of the [SliverLayoutBuilder] builder function.
+typedef SliverLayoutWidgetBuilder = Widget Function(BuildContext context, SliverConstraints constraints);
```

Comment:
You could have a `typedef ConstrainedLayoutBuilder = Widget Function<T extends Constraints>(BuildContext context, T constraints);` for the abstract class, and then I think this all works.

Reply:
We shouldn't get rid of the existing typedef, but maybe we mark it deprecated in favor of the new generic version.